### PR TITLE
nixos/dovecot: add an option to enable mail_plugins

### PR DESCRIPTION
**Pull request by symphorien**
Motivation:
if enableQuota is true, mail plugins cannot be enabled in extraConfig
because of the problem described here:

https://doc.dovecot.org/configuration_manual/config_file/config_file_syntax/#variable-expansion

doveconf: Warning: /etc/dovecot/dovecot.conf line 8: Global setting
mail_plugins won't change the setting inside an earlier filter at
/etc/dovecot/dovecot.conf line 5 (if this is intentional, avoid this
warning by moving the global setting before /etc/dovecot/dovecot.conf
line 5)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @peti @rickynils @fpletz @globin

### CODE REVIEW

**Comment by andir** [comment]
I think a test would be nice that checks that the enabled plugins do not break the configuration files.

**Comment by symphorien** [comment]
I don't really understand what you mean. Using these options in the dovecot test to check they don't introduce syntax errors, or adding a brand new test ?

File: nixos/modules/services/mail/dovecot.nix [hunk 1]
```diff
@@ -183,6 +186,40 @@ in
...
+          };
+        };
+      };
+    in
+    mkOption {
+      type = with types; submodule {
+        options = {
+          globally = mkOption {
+            type = plugins "top-level";
+            example = { enable =[ "virtual" ]; };
```
Comment (Ekleog) [inline nixos/modules/services/mail/dovecot.nix]:
```suggestion
            example = { enable = [ "virtual" ]; };
```

**Review by Ekleog — APPROVED** [review]

**Comment by Ekleog** [comment]
@andir As far as I can read no additional plugins are enabled by-default, which means that by default nothing changes. As such, I don't think an additional test is required? Though the more tests the better, it doesn't feel like something that would necessarily belong here.

**Comment by symphorien** [comment]
fixed the merge conflict

**Pull request merged**